arch/arm64/src/common/arm64_arch_timer.c

```diff
@@ -56,6 +56,7 @@ struct arm64_oneshot_lowerhalf_s
 
   void *arg;                          /* Argument that is passed to the handler */
   uint64_t cycle_per_tick;            /* cycle per tick */
+  uint64_t next_cycle;                /* next cycle start time, i.e. compare register value */
```

Comment:
Actually. To simplify, I don't need to record this value. Just read the counter value every time, and align that to tick boundary.